### ISSUE TITLE
markdown-shifttab breaks

### ISSUE THREAD

**Issue by conal** [jrblevin/markdown-mode#156]
`markdown-shifttab` dies on the following nearly-empty (gitit-style) document:

```
---
title: Another symptom
...

```

Markdown-mode, version 2.1; Emacs version 24.5.1 on Mac OS (from brew).

The backtrace:

```
Debugger entered--Lisp error: (wrong-type-argument integer-or-marker-p nil)
  markdown-outline-fix-visibility()
  markdown-cycle(t)
  markdown-shifttab()
  call-interactively(markdown-shifttab nil nil)
  command-execute(markdown-shifttab)
```

I suspect that this symptom is related to issue #155.

**Comment by syohex** [comment]
I suppose this issue is fixed by #157. Could you check latest version ?

**Comment by conal** [comment]
@syohex Your commit does appear to fix both #157 and #155. Thank you *very* much!

**Comment by syohex** [comment]
Thanks for checking.
